app/assets/javascripts/discourse/app/components/edit-topic-timer-form.js

```diff
@@ -39,26 +40,143 @@ export default Component.extend({
+        timeFormatKey: "dates.long_no_year",
+      },
+      {
+        icon: "far-clock",
+        id: "set_based_on_last_post",
+        label: "topic.auto_update_input.set_based_on_last_post",
+        time: null,
+        timeFormatted: "",
+        hidden: !includeBasedOnLastPost,
+      },
+    ];
+  },
+
+  @discourseComputed
+  hiddenTimeShortcutOptions() {
+    return ["none", "start_of_next_business_week"];
+  },
+
+  isCustom: equal("timerType", "custom"),
+  isBasedOnLastPost: equal("timerType", "set_based_on_last_post"),
+  includeBasedOnLastPost: equal("statusType", CLOSE_STATUS_TYPE),
+
+  @discourseComputed(
+    "topicTimer.updateTime",
+    "topicTimer.duration",
+    "useDuration",
+    "durationType"
+  )
+  executeAt(updateTime, duration, useDuration, durationType) {
+    if (useDuration) {
+      return moment().add(parseFloat(duration), durationType).format(FORMAT);
+    } else {
+      return updateTime;
+    }
+  },
+
+  @discourseComputed(
+    "isBasedOnLastPost",
+    "topicTimer.duration",
+    "topic.last_posted_at"
+  )
+  willCloseImmediately(isBasedOnLastPost, duration, lastPostedAt) {
+    if (isBasedOnLastPost && duration) {
+      let closeDate = moment(lastPostedAt);
+      closeDate = closeDate.add(duration, "hours");
+      return closeDate < moment();
+    }
+  },
+
+  @discourseComputed("isBasedOnLastPost", "topic.last_posted_at")
+  willCloseI18n(isBasedOnLastPost, lastPostedAt) {
+    if (isBasedOnLastPost) {
+      const diff = Math.round(
+        (new Date() - new Date(lastPostedAt)) / (1000 * 60 * 60)
+      );
+      return I18n.t("topic.auto_close_momentarily", { count: diff });
     }
   },
 
-  didReceiveAttrs() {
-    this._super(...arguments);
+  @discourseComputed("durationType")
+  durationTranslationKey(durationType) {
+    return `topic.topic_status_update.num_of_${durationType}`;
+  },
+
+  @discourseComputed(
+    "statusType",
+    "isCustom",
+    "topicTimer.updateTime",
+    "willCloseImmediately",
+    "topicTimer.category_id",
+    "useDuration",
+    "topicTimer.duration"
+  )
+  showTopicStatusInfo(
+    statusType,
+    isCustom,
+    updateTime,
+    willCloseImmediately,
+    categoryId,
+    useDuration,
+    duration
+  ) {
+    if (!statusType || willCloseImmediately) {
+      return false;
+    }
+
+    if (statusType === PUBLISH_TO_CATEGORY_STATUS_TYPE && isEmpty(categoryId)) {
+      return false;
+    }
 
-    // TODO: get rid of this hack
-    schedule("afterRender", () => {
-      if (!this.get("topicTimer.status_type")) {
-        this.set(
-          "topicTimer.status_type",
-          this.get("timerTypes.firstObject.id")
-        );
+    if (isCustom && updateTime) {
+      if (moment(updateTime) < moment()) {
+        return false;
       }
-    });
+    } else if (useDuration) {
+      return duration;
+    }
+
+    return updateTime;
+  },
+
+  @action
+  onTimeSelected(type, time) {
+    this.set("timerType", type);
```

Comment:
group these with setProperties

Reply:
@jjaffeux I tried but I could not make this work, what is the correct way if there is a `.` in the key name:

```
this.setProperties({
  "topicTimer.based_on_last_post": type === "set_based_on_last_post,
  timerType: type
});
```

Or is that correct and maybe I just had some other issue?